lib/core-classes/PayloadService.ts

```diff
@@ -151,45 +151,61 @@ export class PayloadService {
       throw new BadRequestError(`Device ${device._id} is not allowed for registration.`);
     }
 
-    const ret = await this.register(device, decoder, request, { refresh });
+    const hasContent = catalogEntry && catalogEntry.content && catalogEntry.content.tenantId;
```

Comment:
Nitpicking: IMHO naming is unclear, this variable is true if a tenant is defined in the catalog entry